invenio_communities/assets/semantic-ui/js/invenio_communities/requests/requests_items/MobileRequestsItem.js

```diff
@@ -5,31 +5,31 @@
 // under the terms of the MIT License; see LICENSE file for more details.
 
 import { i18next } from "@translations/invenio_app_rdm/i18next";
-import _get from "lodash/get";
-import _truncate from "lodash/truncate";
 import React from "react";
 import RequestTypeLabel from "@js/invenio_requests/request/RequestTypeLabel";
 import RequestStatusLabel from "@js/invenio_requests/request/RequestStatusLabel";
 import { RequestActionController } from "@js/invenio_requests/request/actions/RequestActionController";
 import { Icon, Item } from "semantic-ui-react";
 import { RightBottomLabel } from "./RightBottomLabel";
+import PropTypes from "prop-types";
+import { requestsResultProps } from "./RequestsResultProps";
 
 export const MobileRequestsItem = ({
   result,
   community,
-  differenceInDays,
-  isCreatorCommunity,
-  creatorName,
   updateQueryState,
   currentQueryState,
 }) => {
-  const refreshAfterAction = () => {
-    updateQueryState(currentQueryState);
-  };
+  const {
+    differenceInDays,
+    isCreatorCommunity,
+    creatorName,
+    refreshAfterAction,
+  } = requestsResultProps(result, updateQueryState, currentQueryState);
```

Comment:
Good job on reusing logic! 

Suggestion: I think that maybe the name can be more descriptive, something like `parseRequestsItemProps`.